wit/streams.wit

```diff
@@ -19,62 +17,64 @@ interface streams {
     enum stream-status {
         /// The stream is open and may produce further data.
         open,
-        /// The stream has ended and will not produce any further data.
+        /// When reading, this indicates that the stream will not produce
+        /// further data.
+        /// When writing, this indicates that the stream will no longer be read.
+        /// Further writes are still permitted.
```

Comment:
The write end of a stream is a mechanism for the reader to tell the writer that it won't be reading any more input. The writer is still welcome to send input, but it will be dropped because no reader exists.

In many existing codebases  when the reader doesn't care to read any more input, it may simply stop reading and won't close the stream. So, there is never a guarantee that something written ends up being read - we are exposing this as a mechanism to allow optimizing the writer.

libc might choose to translate this into EPIPE when appropriate, or even emulate a SIGPIPE exit when appropriate (e.g. on stdin). (I'm not a libc expert but those are Dan's suggestions.)

This feels to me like a weaker condition than an error, so I think its a case where we can be liberal in what we accept - its easy enough to let the writer keep writing and drop its input, if it doesn't honor the flag indicating the stream has closed.

Reply:
All clear to me now. Thanks.

Reply:
We've been having some issues with the write backpressure design and I've come around to the opposite view on this topic... I'm working on the design and implementation of a different approach now which I intend will replace this PR.